### ISSUE TITLE
Poor performance of newly-trained model

### ISSUE THREAD

**Issue by Jiale-Fan** [jchengai/planTF#6]
Hi, thank you for making your code public first of all!

I was trying to train the model totally following instructions in the readme file and hyperparameters in the paper, but it turns out my trained model performs a lot poorer than the pre-trained model given by you. Here is the command I used to train the model, the performance comparison, and the wandb page of the training process. I would appreciate very much if you can help to check if there is anything wrong of my training process. 

`
export CUDA_VISIBLE_DEVICES=0,1,2
python run_training.py \
  py_func=train +training=train_planTF \
  worker=single_machine_thread_pool worker.max_workers=16 \
  scenario_builder=nuplan cache.cache_path=/data1/nuplan/jiale2/exp/cache_plantf_1M cache.use_cache_without_dataset=true \
  data_loader.params.batch_size=128 data_loader.params.num_workers=8 \
  data_loader.datamodule.train_fraction=1.0 \
  data_loader.datamodule.val_fraction=0.01 \
  data_loader.datamodule.test_fraction=0.01 \
  lr=1e-3 epochs=35 warmup_epochs=3 weight_decay=0.0001 \
  lightning.trainer.params.check_val_every_n_epoch=4 \
  wandb.mode=online wandb.project=nuplan wandb.name=plantf
`

  | Hard OLS | Hard CLS-NR | Hard CLS-R | Random OLS | Random CLS-NR | Random CLS-R
-- | -- | -- | -- | -- | -- | --
Paper reports | 83.32 | 72.68 | 61.7 | 87.07 | 86.48 | 80.59
Our simulation | (interrupted)  | 47.66 | 45.85 | 47.88 | 49.51 | 52.99

wandb page: https://wandb.ai/jialef22/nuplan?workspace=user-jialef22

Also, may I know how do you determine which checkpoint file is used for simulation? The last checkpoint or the checkpoint with least validation error?

Thank you in advance
Best

**Comment by jchengai** [comment]
Hi, @Jiale-Fan, `data_loader.params.batch_size=128`  is the batch size of each GPU, which means you use a total batch size of 128×3 (Your command use 3 GPUs). Besides, `data_loader.datamodule.val_fraction=0.01` corresponds to 800 validation frames, which is too small. This following command is the training setting reported in the paper:

```
CUDA_VISIBLE_DEVICES=0,1,2,3 python run_training.py \
  py_func=train +training=train_planTF \
  worker=single_machine_thread_pool worker.max_workers=32 \
  scenario_builder=nuplan cache.cache_path=/nuplan/exp/cache_plantf_1M cache.use_cache_without_dataset=true \
  data_loader.params.batch_size=32 data_loader.params.num_workers=32 \
  lr=1e-3 epochs=25 warmup_epochs=3 weight_decay=0.0001 \
  lightning.trainer.params.val_check_interval=0.5 \
  wandb.mode=online wandb.project=nuplan wandb.name=plantf
``` 

![image](https://github.com/jchengai/planTF/assets/86758357/58342768-59af-477b-974d-104a4d880558)

**Comment by jchengai** [comment]
> Also, may I know how do you determine which checkpoint file is used for simulation? The last checkpoint or the checkpoint with least validation error?

Empirically, I found there is no clear correlation between the open-loop validation performance and closed-loop testing performance. In practice, you may choose the checkpoint by conducting a small set of closed-loop simulations on the validation set.

**Comment by Jiale-Fan** [comment]
Thank you for your reply! But I still wonder if a very small validation set would affect training results? I think the size of validation set and validation frequency are only related to metrics computing and would not impact or alter the way the model is trained in any way.

**Comment by Jiale-Fan** [comment]
Hi, I notice during the training there is following output, indicating that the 1 million samples are not all for training; since the nuplan devkit splits the the dataset into train set and validation set according to the config file `nuplan.yaml` which initiating `nuplan.planning.training.data_loader.log_splitter.LogSplitter`, the number of samples for training is about 330k, and is not 1M. Could you confirm that this is the way you intend to train, instead of using all 1M scenarios as train set?

`
2023-10-30 21:52:20,924 INFO {/home/jiale/planTF/src/custom_training/custom_datamodule.py:56}  Number of samples in train set: 331299
2023-10-30 21:52:20,925 INFO {/home/jiale/planTF/src/custom_training/custom_datamodule.py:56}  Number of samples in train set: 331299
2023-10-30 21:52:20,975 INFO {/home/jiale/planTF/src/custom_training/custom_datamodule.py:56}  Number of samples in train set: 331299
2023-10-30 21:52:21,508 INFO {/home/jiale/planTF/src/custom_training/custom_datamodule.py:56}  Number of samples in validation set: 668701
2023-10-30 21:52:21,510 INFO {/home/jiale/planTF/src/custom_training/custom_datamodule.py:56}  Number of samples in validation set: 668701
2023-10-30 21:52:21,553 INFO {/home/jiale/planTF/src/custom_training/custom_datamodule.py:56}  Number of samples in validation set: 668701
`

**Comment by jchengai** [comment]
This looks strange. I believe that the train-val ratio should be 9:1. FYI, that's my output

```
2023-10-31 01:42:50,135 INFO {/workspace/planTF/src/custom_training/custom_datamodule.py:56}  Number of samples in train set: 919925
2023-10-31 01:42:50,304 INFO {/workspace/planTF/src/custom_training/custom_datamodule.py:56}  Number of samples in train set: 919925
2023-10-31 01:42:50,383 INFO {/workspace/planTF/src/custom_training/custom_datamodule.py:56}  Number of samples in train set: 919925
2023-10-31 01:42:50,414 INFO {/workspace/planTF/src/custom_training/custom_datamodule.py:56}  Number of samples in validation set: 80075
2023-10-31 01:42:50,459 INFO {/workspace/planTF/src/custom_training/custom_datamodule.py:56}  Number of samples in train set: 919925
2023-10-31 01:42:50,582 INFO {/workspace/planTF/src/custom_training/custom_datamodule.py:56}  Number of samples in validation set: 80075
2023-10-31 01:42:50,664 INFO {/workspace/planTF/src/custom_training/custom_datamodule.py:56}  Number of samples in validation set: 80075
2023-10-31 01:42:50,750 INFO {/workspace/planTF/src/custom_training/custom_datamodule.py:56}  Number of samples in validation set: 80075
```

**Comment by Jiale-Fan** [comment]
Thank you for the information. I think I must have some problem with the configuration of my dataset. I think the 1M scenarios you use to train the model should come from all following divisions of the "nuPlan train split"?
![image](https://github.com/jchengai/planTF/assets/68210288/9883e99c-d74a-4eb0-96b5-fe6bbc54c336)

**Comment by jchengai** [comment]
Yes, I use the entire training set from 4 cities.
